File: web/src/lib/sentiment.ts
export const SENTIMENTS = [
  "Hostile",
  "Tense",
  "Frustrated",
  "Skeptical",
  "Neutral",
  "Engaged",
  "Positive",
  "Enthusiastic",
] as const;

export type Sentiment = (typeof SENTIMENTS)[number];

export function parseSentiment(raw: unknown): Sentiment | null {
  if (typeof raw !== "string") return null;
  const needle = raw.trim().toLowerCase();
  return SENTIMENTS.find((value) => value.toLowerCase() === needle) ?? null;
}

/** Discrete fill 0..1 for the live gauge — Hostile empty, Enthusiastic full. */
export function sentimentFill(value: Sentiment): number {
  return SENTIMENTS.indexOf(value) / (SENTIMENTS.length - 1);
}

File: server/app/services/copilot/sentiment.py
"""Closed sentiment vocabulary for live copilot and post-call reports.

Tone (warmth), not outcome — coach_score is how well the call is going
for Me. Unknown / free-text LLM output becomes None rather than Neutral
so analysis is not poisoned by a dump bucket.
"""

SENTIMENTS: tuple[str, ...] = (
    "Hostile",
    "Tense",
    "Frustrated",
    "Skeptical",
    "Neutral",
    "Engaged",
    "Positive",
    "Enthusiastic",
)

_BY_LOWER = {value.lower(): value for value in SENTIMENTS}

# Exact string the live/report prompts quote so the model cannot invent
# "Warm" / "Mixed" / "Receptive".
SENTIMENT_CHOICES = ", ".join(SENTIMENTS)


def parse_sentiment(raw: object) -> str | None:
    """Canonical enum value, or None if missing / not in the closed set."""
    if not isinstance(raw, str):
        return None
    return _BY_LOWER.get(raw.strip().lower())
